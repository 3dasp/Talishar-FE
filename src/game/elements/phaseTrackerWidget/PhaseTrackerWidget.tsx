import React from 'react';
import styles from './PhaseTrackerWidget.module.css';

export default function PhaseTrackerWidget() {
  //TODO: Add functionality to this.
  return <div className={styles.phaseWidget}></div>;
}
