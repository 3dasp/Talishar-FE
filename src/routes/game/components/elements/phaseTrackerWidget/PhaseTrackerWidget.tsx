import useWindowDimensions from 'hooks/useWindowDimensions';
import React from 'react';
import styles from './PhaseTrackerWidget.module.css';

export interface TurnPhase {
  phase?: string;
}

export default function PhaseTrackerWidget({ phase }: TurnPhase) {
  const [windowWidth, windowHeight] = useWindowDimensions();

  // height of component needs to be 9vh;
  const scaleFactor = 0.09 * (windowHeight / 85);
  let styleToApply = { left: '0em' };

  console.log(scaleFactor);

  switch (phase) {
    case 'M':
      styleToApply = { left: '8.5px' };
      break;
    case 'P':
      styleToApply = { left: '45.5px' };
      break;
    case 'B':
      styleToApply = { left: '74px' };
      break;
    case 'A':
      styleToApply = { left: '102.5px' };
      break;
    case 'D':
      styleToApply = { left: '102.5px' };
      break;
    case 'ARS':
      styleToApply = { left: '139.5px' };
      break;
    case 'PDECK':
      styleToApply = { left: '139.5px' };
      break;
    default:
      styleToApply = { left: '8.5px' };
      break;
  }

  return (
    <div className={styles.notScaled}>
      <div
        className={styles.phaseWidget}
        style={{ transform: `scale(${scaleFactor})` }}
      >
        <div className={styles.icon} style={styleToApply}></div>
      </div>
    </div>
  );
}
