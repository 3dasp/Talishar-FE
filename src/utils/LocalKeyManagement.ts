export const saveGameAuthKey = (gameId: number, authKey: string): void => {
  if (gameId <= 0) {
    console.error('Invalid game ID, cannot save authKey for game:', gameId);
    return;
  }
  sessionStorage.setItem(String(gameId), authKey);
};

export const loadGameAuthKey = (gameId: number): string => {
  if (gameId <= 0) {
    console.error('Invalid game ID, cannot load authKey for game:', gameId);
    return '';
  }

  const localValue = sessionStorage.getItem(String(gameId));
  if (!localValue) {
    console.error(
      'Attempted to access an authKey of a game that does not exist, game:' +
        gameId
    );
  }
  return localValue || '';
};

export const deleteGameAuthKey = (gameId: number): void => {
  if (gameId <= 0) {
    console.error('Invalid game ID, cannot delete authKey for game:', gameId);
    return;
  }

  const valueExists = sessionStorage.getItem(String(gameId));
  if (!valueExists) {
    console.error(
      'Attempted to delete a authKey of a game that does not exist, game',
      gameId
    );
    return;
  }
  sessionStorage.removeItem(String(gameId));
};
