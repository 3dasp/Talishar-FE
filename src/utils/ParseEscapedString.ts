const CARDRE = /{{(.+?)\|(.+?)\|(.+?)}}/g;

export const replaceText = (inputString: string): string => {
  const mapping: { [key: string]: string } = {
    '0': 'gray',
    '1': 'red',
    '2': 'yellow',
    '3': 'blue'
  };

  return inputString.replace(CARDRE, (match, p1, p2, p3) => {
    const color = mapping[p3];
    const imgPath = `./WebpImages/${p1}.webp`;
    return `<span onmouseover="ShowDetail(event, '${imgPath}')" onmouseout="HideCardDetail()" style="color:${color}">${p2}</span>`;
  });
};
