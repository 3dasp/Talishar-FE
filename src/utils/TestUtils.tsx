/* eslint-disable @typescript-eslint/ban-types */
import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { AppStore, RootState, setupStore } from '../app/Store';

import InitialGameState from '../features/game/InitialGameState';
import { MemoryRouter } from 'react-router-dom';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    // @ts-ignore - I know I know...
    preloadedState = {
      game: InitialGameState
    },
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <MemoryRouter>{children}</MemoryRouter>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};
