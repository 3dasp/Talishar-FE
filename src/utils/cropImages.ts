import { CLOUD_IMAGES_URL } from "appConstants";

export const generateCroppedImageUrl = (cardName: string): string => `${CLOUD_IMAGES_URL}/crops/${cardName}_cropped.png`;