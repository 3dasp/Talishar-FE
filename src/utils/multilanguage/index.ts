export {
  getCollectionCardImagePath,
  getCardBackImagePath,
  loadInitialLanguage
} from './multilanguage';
export type { pathType } from './types';
export {
  DEFAULT_LANGUAGE,
  CARD_IMAGES_PATH,
  CARD_SQUARES_PATH
} from './constants';
