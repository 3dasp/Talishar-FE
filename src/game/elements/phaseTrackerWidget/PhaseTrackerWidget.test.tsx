import React from 'react';
import { renderWithProviders } from '../../../utils/TestUtils';
import PhaseTrackerWidget from './PhaseTrackerWidget';

it('matches snapshot main phase', () => {
  const div = renderWithProviders(<PhaseTrackerWidget phase={'M'} />);
  const display = document.querySelector('div');
  expect(display).toMatchSnapshot();
});

it('matches snapshot defence reaction phase', () => {
  const div = renderWithProviders(<PhaseTrackerWidget phase={'D'} />);

  const display = document.querySelector('div');
  expect(display).toMatchSnapshot();
});
