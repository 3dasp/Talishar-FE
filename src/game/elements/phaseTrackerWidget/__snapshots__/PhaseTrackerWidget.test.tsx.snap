// Vitest Snapshot v1

exports[`matches snapshot defence reaction phase 1`] = `
<div>
  <div
    class="_phaseWidget_58c017"
  >
    <div
      class="_icon_58c017"
      style="left: 7.08em;"
    />
  </div>
</div>
`;

exports[`matches snapshot main phase 1`] = `
<div>
  <div
    class="_phaseWidget_58c017"
  >
    <div
      class="_icon_58c017"
      style="left: 0.42em;"
    />
  </div>
</div>
`;
